Turn off scalar_checks for torch.take.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30438 Turn off scalar_checks for SpatialDepthwiseConvolution and SpatialConvolutionMM.
* #30437 Remove scalar_check from topk, move it to the THC implementation.
* #30436 Turn off scalar_checks for _th_reciprocal.
* #30435 Turn off scalar_checks for torch.clamp.
* #30434 Turn off scalar_checks for exp, cos, cosh, tan, atan, tanh, erf, erfc.
* #29955 [WIP] Turn off scalar_check for _th_normal.
* #29954 Turn off scalar_check for _th_gather.
* **#29953 Turn off scalar_checks for torch.take.**
* #29952 Turn off scalar_check for Tensor.set_(Tensor)

The underlying function handles it correctly.

Differential Revision: [D18548055](https://our.internmc.facebook.com/intern/diff/D18548055)

gchanan committed:
Turn off scalar_checks for torch.take.

The underlying function handles it correctly.

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.take."


The underlying function handles it correctly.

Differential Revision: [D18548055](https://our.internmc.facebook.com/intern/diff/D18548055)

[ghstack-poisoned]

albanD reviewed (approved):
nit: some linting in the test

gchanan committed:
Update on "Turn off scalar_checks for torch.take."


The underlying function handles it correctly.

Differential Revision: [D18548055](https://our.internmc.facebook.com/intern/diff/D18548055)

[ghstack-poisoned]

gchanan committed:
Update on "Turn off scalar_checks for torch.take."


The underlying function handles it correctly.

Differential Revision: [D18548055](https://our.internmc.facebook.com/intern/diff/D18548055)

[ghstack-poisoned]
